mediaire_toolbox/transaction_db/transaction_db.py

```diff
@@ -484,7 +470,7 @@ def unset_patient_consent(self, id_: int):
         """Mark this transaction ID with NO data usage patient consent"""
         try:
             t = self._get_transaction_or_raise_exception(id_)
-            t.patient_consent = 0
+            t.patient_consent_date = None
```

Comment:
@phistep I think the safest and more backwards compatible thing here would be to keep both columns and their semantic too, intact. I don't know now if somewhere somehow we will only check for the patient_consent column which wouldn't be properly updated after this code is merged. Since the update in the database is atomic, I think it's fine setting both patient_consent = 1 and patient_consent_date to the date, or unsetting them at the same time (also atomically).

Reply:
It would only be a problem, if someone is accessing the sqlite directly. The @property takes care of providing the old interface to the new column transparently. The old column would just be dead and not deleted only for the reason of simplifying the migration.

Even tough I think it is a bad idea to keep both fields, it will just lead to confusion, I accept you concern. I reintroduced the column, but marked the ORM field as private, so it cannot be accessed directly. The @property will take care of updating both columns now, but the date will be the only source of truth. The boolean is only updated to keep compatibility with tools not using the ORM but accessing the SQLite directly.
Using [`@hybrid_property`](https://docs.sqlalchemy.org/en/12/orm/extensions/hybrid.html) enables using `patient_consent` in query chains, as the internal expression will be translated to SQL.

Reply:
So my idea was rather to keep compatibility for direct SQL usage and even older version so the toolbox, but I understand your point of not wanting to maintain two columns with identical (derivable) meaning.
The Mixin is a great piece of engineering and I was excited to find out that one can do such a thing. Seeing the amount of complexity introduced, do we really want to introduce it though? I am willing to compromise on the idea of making it backwards compatible so up to you: either get rid of all this and go on with the non-backwards compatible change, or keep the Mixin.

Reply:
Haha okay it seems like we both took the other persons stance now :D We should definitely keep the API backwards compatible, there is just so much code out there that depends on it. But even that covers most of the complexity introduced in the Mixin, since we need to do the proxy getters/setters regardless. And then I guess it doesn't hurt that much to go all the way. consider the fact that we would probably need to change a lot of metabase queries as well. Sure one could also mock the interface there ```SELECT (patient_consent_date IS NOT NULL) AS patient_consent```, but that's also a lot of work...

I'd say we keep it. It's pretty well isolated from the rest, even if the internals have complexity. If we decide to remove the bool field in the future, it's easy enough to do. The other way round is going to be much bigger problem.

Reply:
Alright, sounds good, we'll just have to carefully assert that it works as expected (as seen by the conversation in anonymizer) - let's continue there